README.md
# Direct Mail

The Direct Mail implements advanced newsletter mailer system for the [TYPO3 CMS](http://typo3.org). This extension provides not only a system for sending a newsletter from TYPO3 CMS, but also personalizing it and collect some statistics.

The extension creates and send a newsletter based on TYPO3 page, so that you can create newsletter layout as you create website layout.

## Documentation and support

-  **Documentation**

   https://docs.typo3.org/typo3cms/extensions/direct_mail/


-  **Slack channel**

   https://typo3.slack.com/messages/direct_mail
   
   To join the slack channel, please request for invitation from https://forger.typo3.org/slack

## Contributing

You can contribute to this project by reporting any issue you found here in github or on https://forge.typo3.org/projects/extension-direct_mail/

You can also contributing any code by:

1. Fork the repository on Github
2. Clone repository
3. Make your changes
4. Commit your changes to your fork. In your commit message refer to the issue number if there is already one, e.g. `[BUGFIX] short description of fix (resolves #4711)`
5. Submit a Pull Request using GitHub (here are some hints on [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request))

ChangeLog
2014-06-12  fa2ae77             [RELEASE] Release of direct_mail 4.0.0 (Ivan Kartolo)
2013-12-16  00a23fe  #54462     [TASK] Code cleanup in CLI script (Bernhard Kraft)
2014-06-04  56c597a  #59342     [BUGFIX] Uploading CSV file fails in importer (Ivan Kartolo)
2013-09-12  76f2c5f  #50527     [TASK] replacing t3lib with namespaces (Ivan Kartolo)
2013-10-11  30b3dfd  #49453     [BUGFIX] Adjust returnmail.phpshb for TYPO3 6.x (Stefan Neufeind)
2013-10-02  913c7ac  #52458     [BUGFIX] Wrong type label in direct mail listing (Bernhard Kraft)
2013-10-28  950f128  #52094     [BUGFIX] initializeHookObjects: array not properly checked (Bernhard Kraft)
2013-12-16  9b493ba  #54461     [BUGFIX] Can't run CLI script (Bernhard Kraft)
2013-12-13  0796fcd  #53173     [BUGFIX] Reapply: Fix assignment in class.dmailer.php (Stefan Neufeind)
2013-12-16  bd7b656             Revert "[BUGFIX] Fix assignment in class.dmailer.php" (Bernhard Kraft)
2013-12-13  f17270b  #53173     [BUGFIX] Fix assignment in class.dmailer.php (Stefan Neufeind)
2013-08-13  9241a00  #52333     [FEATURE] Allow to disable jumpurl for single links (Bernhard Kraft)
2013-08-28  fe4f335             [BUGFIX] security fix: jumpurl and information disclosure (Ivan Kartolo)
2013-08-28  2a28260             Revert "[BUGFIX] security fix: jumpurl and information disclosure" (Ivan Kartolo)
2013-08-28  a656469             [BUGFIX] security fix: jumpurl and information disclosure (Ivan Kartolo)
2013-09-12  84bba56  #51928     [TASK] Removing gabriel task (Ivan Kartolo)
2013-08-21  7913be1  #46779     [TASK] Avoid using loadTCA (Ivan Kartolo)
2013-09-04  fe21229  #44725     [BUGFIX] Importer is not compatible with TYPO3 6.x with FAL (Saverio Vigni)
2013-07-30  4e69d05             Merge: 53f5e28 8858859 (Ivan Kartolo)
2013-07-29  53f5e28             [TASK] Setting version to 4.0.0-dev (Ivan Kartolo)
2013-08-28  85b34b9             [BUGFIX] security fix: jumpurl and information disclosure (Ivan Kartolo)
2013-07-30  42f959d             [RELEASE] Release of 3.1.1 (Ivan Kartolo)
2013-07-30  eff44c1             [TASK] add dependency of TYPO3 version (Ivan Kartolo)
2013-07-29  d2c3ccb             [RELEASE] Release of direct_mail 3.1.0 (Ivan Kartolo)
2013-07-29  5e4bb80             [BUGFIX] fixes SQL definition for sys_dmail_maillog table (Ivan Kartolo)
2013-07-26  f4b217b  #48035     [BUGFIX] sql_free_result not working with bool (Ivan Kartolo)
2013-07-25  e777918             [TASK] set version to 3.1.0-dev (Ivan Kartolo)
2013-07-18  1207ce3  #50053     [BUGFIX] dmailerping is not correctly counted (Ivan Kartolo)
2013-07-08  0d63c2d  #48889     [BUGFIX] Fix possible jumpurl/dmailerping security leak (Bernhard Kraft)
2013-06-10  7a87231  #46454     [BUGFIX] Plain text link counters don't work (Bernhard Kraft)
2013-06-10  ee08af1  #48994     [FEATURE] Adding hooks when sending direct mails via scheduler (Bernhard Kraft)
2013-06-12  d5ec09d  #48727     [FEATURE] Improve privacy of newsletter jumpurl links (Bernhard Kraft)
2013-06-27  6e80e36  #49422     [FEATURE] ###XID### replacement in return_path (Ivan Kartolo)
2013-06-24  fa1bb9b  #49378     [FEATURE] Trim items of address list readed from csv (Frans Saris)
2013-06-16  d1ad346  #48515     [BUGFIX] Trim email addresses when fetching from database (Ivan Kartolo)
2013-06-10  5988d55  #46452     [BUGFIX] Statistics module uses unnecessary t3lib_div::inList (Bernhard Kraft)
2013-06-04 Ivan Kartolo <ivan@kartolo.de>
	* Feature #48723: added hide field in importer mapping

2013-05-09 Bernhard Kraft <kraft@webconsulting.at>
	* Bug #48088: URLs returned by "tx_directmail_static::getUrlBase()" were wrong when called via the scheduler

2013-05-06 Ivan Kartolo <ivan@kartolo.de>
	* Bug #46224: check if allow_url_fopen is active, else use getUrl
	* do not check the whole path for the dmailerping
	* Bug #46226: added first_name, middle_name and last_name to the recipient list exporter

2013-03-11 Ivan Kartolo <ivan@kartolo.de>
	* Bug #46087: Fixed how the jumpurl is redirected to TYPO3. Thanks to Peter Niederlag
	* Release 3.0.4

2013-01-21 Ivan Kartolo <ivan@kartolo.de>
	* Bug #44579: Fixed date picker in the wizard.

2013-01-18 Ivan Kartolo <ivan@kartolo.de>
	* Bug #44588: updated TCA

2012-12-27 Ivan Kartolo <ivan@kartolo.de>
	* Fixed typo in extension config
	* Release 3.0.3

2012-12-11 Ivan Kartolo <ivan@kartolo.de>
	* Bug #43064: check if array is given in get_localized_categories userFunc.
	* Bug #42544: bug fixed wrong link scheme if TYPO3 BE is https

2012-12-09 Ivan Kartolo <iva@kartolo.de>
	* Bug #41340: fixed recipient header if no name is given
	* Bug #41480: message ID uses the host name

2012-12-05 Ivan Kartolo <ivan@kartolo.de>
	* Bug #38905: background image from table, tr or td tag can now be embedded in the newsletter
	* Bug #41536: bug fix on using the domain for internal links

2012-11-11 Ivan Kartolo <ivan@kartolo.de>
	* Bug #39883: using getURL to fetch attachment and embedded media
	* Bug #42107: don't set ReplyTo if it's empty

2012-10-04 Ivan Kartolo <ivan@kartolo.de>
	* Bug #39422: fixed dmailerping not counted in statistic module

2012-07-11 Ivan Kartolo <ivan@kartolo.de>
	* Bug #37781: added bounce reason
	* Bug #37989: fixed file operation permission for non-admin in importer

2012-07-04 Ivan Kartolo <ivan@kartolo.de>
	* Bug #38003: hide query editor if user doesn't have right to the recipient table
	* Bug #38232: fixed fetching included media from the wrong FE path

2012-07-01 Ivan Kartolo <ivan@kartolo.de>
	* Bug #38479: fixed recursively added media (thx to Tom)
	* Bug #38003: remove the edit link of recipient list, if the user doesn't have the right to the table

2012-06-08 Ivan Kartolo <ivan@kartolo.de>
	* Bug #37821: fixed utf-8 problem in subject

2012-06-06 Ivan Kartolo <ivan@kartolo.de>
	* remove deprecated dmailer.phpsh

2012-06-04 Ivan Kartolo <ivan@kartolo.de>
	* Bug #31054: fix compiling recipient from fe_groups. module_sys_dmail_newsletter was ignored


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "direct_mail".
 *
 * Auto generated 29-07-2013 16:04
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Direct Mail',
	'description' => 'Advanced Direct Mail/Newsletter mailer system with sophisticated options for personalization of emails including response statistics.',
	'category' => 'module',
	'shy' => 0,
	'version' => '4.0.0',
	'dependencies' => 'cms,tt_address',
	'conflicts' => 'sr_direct_mail_ext,it_dmail_fix,plugin_mgm,direct_mail_123',
	'priority' => '',
	'loadOrder' => '',
	'module' => 'mod1,mod2,mod3,mod4,mod5,mod6',
	'state' => 'stable',
	'uploadfolder' => 1,
	'createDirs' => '',
	'modify_tables' => 'tt_content,tt_address,fe_users',
	'clearcacheonload' => 0,
	'lockType' => '',
	'author' => 'Ivan Kartolo',
	'author_email' => 'ivan.kartolo@dkd.de',
	'author_company' => 'd.k.d Internet Service GmbH',
	'CGLcompliance' => '',
	'CGLcompliance_note' => '',
	'constraints' => array(
		'depends' => array(
			'cms' => '',
			'tt_address' => '',
			'php' => '5.3.0',
			'typo3' => '6.2.2-6.2.99',
		),
		'conflicts' => array(
			'sr_direct_mail_ext' => '',
			'it_dmail_fix' => '',
			'plugin_mgm' => '',
			'direct_mail_123' => '',
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:99:{s:9:"ChangeLog";s:4:"c147";s:20:"class.ext_update.php";s:4:"0ab1";s:31:"class.tx_directmail_gabriel.php";s:4:"6de4";s:33:"class.tx_directmail_scheduler.php";s:4:"1a0e";s:16:"ext_autoload.php";s:4:"2e3e";s:21:"ext_conf_template.txt";s:4:"7c49";s:12:"ext_icon.gif";s:4:"a143";s:17:"ext_localconf.php";s:4:"33b2";s:14:"ext_tables.php";s:4:"bc0a";s:14:"ext_tables.sql";s:4:"2388";s:17:"locallang_tca.xml";s:4:"6e8b";s:35:"Classes/Scheduler/MailFromDraft.php";s:4:"4f4c";s:52:"Classes/Scheduler/MailFromDraft_AdditionalFields.php";s:4:"eaa9";s:21:"Configuration/tca.php";s:4:"0000";s:42:"Interfaces/Scheduler/MailFromDraftHook.php";s:4:"938b";s:40:"Resources/Public/StyleSheets/modules.css";s:4:"bf1f";s:23:"cli/cli_direct_mail.php";s:4:"a2b3";s:14:"doc/manual.sxw";s:4:"811a";s:36:"locallang/locallang_csh_sysdmail.xml";s:4:"f4a1";s:39:"locallang/locallang_csh_sysdmailcat.xml";s:4:"a2b1";s:37:"locallang/locallang_csh_sysdmailg.xml";s:4:"0ecd";s:42:"locallang/locallang_csh_txdirectmailM2.xml";s:4:"2fa2";s:42:"locallang/locallang_csh_txdirectmailM3.xml";s:4:"3846";s:42:"locallang/locallang_csh_txdirectmailM4.xml";s:4:"761f";s:42:"locallang/locallang_csh_txdirectmailM5.xml";s:4:"e511";s:42:"locallang/locallang_csh_txdirectmailM6.xml";s:4:"3f6d";s:44:"locallang/locallang_csh_web_txdirectmail.xml";s:4:"1764";s:30:"locallang/locallang_mod2-6.xml";s:4:"d14a";s:14:"mod1/clear.gif";s:4:"cc11";s:13:"mod1/conf.php";s:4:"f25a";s:14:"mod1/index.php";s:4:"4f8b";s:22:"mod1/locallang_mod.xml";s:4:"9b3c";s:17:"mod1/mod_icon.gif";s:4:"a143";s:22:"mod1/mod_template.html";s:4:"65bd";s:34:"mod2/class.tx_directmail_dmail.php";s:4:"5beb";s:13:"mod2/conf.php";s:4:"f24c";s:14:"mod2/index.php";s:4:"6421";s:22:"mod2/locallang_mod.xml";s:4:"6088";s:17:"mod2/mod_icon.gif";s:4:"a143";s:22:"mod2/mod_template.html";s:4:"f729";s:43:"mod3/class.tx_directmail_recipient_list.php";s:4:"7ad3";s:14:"mod3/clear.gif";s:4:"cc11";s:13:"mod3/conf.php";s:4:"ba64";s:14:"mod3/index.php";s:4:"e742";s:22:"mod3/locallang_mod.xml";s:4:"c2ce";s:17:"mod3/mod_icon.gif";s:4:"a143";s:22:"mod3/mod_template.html";s:4:"2581";s:39:"mod4/class.tx_directmail_statistics.php";s:4:"95da";s:14:"mod4/clear.gif";s:4:"cc11";s:13:"mod4/conf.php";s:4:"2c51";s:14:"mod4/index.php";s:4:"e2b7";s:22:"mod4/locallang_mod.xml";s:4:"fc77";s:17:"mod4/mod_icon.gif";s:4:"a143";s:22:"mod4/mod_template.html";s:4:"2581";s:42:"mod5/class.tx_directmail_mailer_engine.php";s:4:"8129";s:14:"mod5/clear.gif";s:4:"cc11";s:13:"mod5/conf.php";s:4:"4ad5";s:14:"mod5/index.php";s:4:"2077";s:22:"mod5/locallang_mod.xml";s:4:"a0d7";s:17:"mod5/mod_icon.gif";s:4:"a143";s:22:"mod5/mod_template.html";s:4:"2581";s:42:"mod6/class.tx_directmail_configuration.php";s:4:"9037";s:14:"mod6/clear.gif";s:4:"cc11";s:13:"mod6/conf.php";s:4:"2862";s:14:"mod6/index.php";s:4:"c58e";s:22:"mod6/locallang_mod.xml";s:4:"87d6";s:17:"mod6/mod_icon.gif";s:4:"a143";s:31:"pi1/class.tx_directmail_pi1.php";s:4:"ef59";s:17:"pi1/locallang.php";s:4:"ff9e";s:17:"pi1/locallang.xml";s:4:"2d6b";s:36:"pi1/tx_directmail_pi1_plaintext.tmpl";s:4:"2027";s:18:"res/gfx/attach.gif";s:4:"5559";s:17:"res/gfx/dmail.gif";s:4:"4d4f";s:22:"res/gfx/dmail_list.gif";s:4:"8d58";s:23:"res/gfx/dmailerping.gif";s:4:"cc11";s:33:"res/gfx/ext_icon_dmail_folder.gif";s:4:"a143";s:39:"res/gfx/icon_tx_directmail_category.gif";s:4:"9398";s:16:"res/gfx/mail.gif";s:4:"4174";s:21:"res/gfx/mailgroup.gif";s:4:"1cc5";s:25:"res/gfx/modules_dmail.gif";s:4:"a143";s:28:"res/gfx/modules_dmail__h.gif";s:4:"040c";s:19:"res/gfx/newmail.gif";s:4:"ffa9";s:24:"res/gfx/preview_html.gif";s:4:"1e65";s:23:"res/gfx/preview_txt.gif";s:4:"4d9a";s:29:"res/scripts/class.dmailer.php";s:4:"4089";s:32:"res/scripts/class.mailselect.php";s:4:"43dd";s:30:"res/scripts/class.readmail.php";s:4:"c526";s:48:"res/scripts/class.tx_directmail_checkjumpurl.php";s:4:"6bf9";s:45:"res/scripts/class.tx_directmail_container.php";s:4:"b13c";s:44:"res/scripts/class.tx_directmail_importer.php";s:4:"6f7e";s:53:"res/scripts/class.tx_directmail_select_categories.php";s:4:"0c1f";s:42:"res/scripts/class.tx_directmail_static.php";s:4:"171f";s:47:"res/scripts/class.tx_directmail_tsparserext.php";s:4:"a5fe";s:52:"res/scripts/class.tx_directmail_ttnews_plaintext.php";s:4:"c28d";s:28:"res/scripts/returnmail.phpsh";s:4:"c0be";s:27:"static/boundaries/setup.txt";s:4:"9409";s:30:"static/plaintext/constants.txt";s:4:"59ce";s:26:"static/plaintext/setup.txt";s:4:"ee48";s:34:"static/tt_news_plaintext/setup.txt";s:4:"1a31";}',
	'suggests' => array(
	),
);

?>